ci-operator/step-registry/aws/deprovision/route53/private-hosted-zone/aws-deprovision-route53-private-hosted-zone-commands.sh

```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+trap 'CHILDREN=$(jobs -p); if test -n "${CHILDREN}"; then kill ${CHILDREN} && wait; fi' TERM
+
+export AWS_SHARED_CREDENTIALS_FILE="${CLUSTER_PROFILE_DIR}/.awscred"
+REGION="${LEASED_RESOURCE}"
+HOSTED_ZONE_ID="${SHARED_DIR}/hosted_zone_id"
+HOSTED_ZONE="${SHARED_DIR}/hosted_zone"
+
+if [ ! -f "${HOSTED_ZONE_ID}" ]; then
+  echo "File ${HOSTED_ZONE_ID} does not exist."
+  exit 1
+fi
+
+echo "Deleting AWS route53 hosted zone"
+aws --region "${REGION}" route53 delete-hosted-zone --id  "$(cat "${HOSTED_ZONE_ID}")" > "${HOSTED_ZONE}"
+CHANGE_ID="$(jq -r '.ChangeInfo.Id' "${HOSTED_ZONE}" | awk -F / '{printf $3}')"
```

Comment:
nit: instead of dropping onto an on-disk `${HOSTED_ZONE}"` file, you can hold this in memory.

```sh
HOSTED_ZONE="$(aws --region "${REGION}" route53 delete-hosted-zone --id  "$(cat "${HOSTED_ZONE_ID}")")"
CHANGE_ID="$(echo "${HOSTED_ZONE}" | jq -r .ChangeInfo.Id | awk -F / '{printf $3}')"
```

Reply:
Done